nav2_localization/src/plugins/solver.cpp

```diff
@@ -0,0 +1,48 @@
+#include "pluginlib/class_list_macros.hpp"
+#include "nav2_localization/interfaces/solver_base.hpp"
+#include "nav2_localization/plugins/solver_plugins.hpp"
+
+namespace nav2_localization
+{
+
+geometry_msgs::msg::PoseWithCovariance DummySolver2d::solve(
+	const nav_msgs::msg::Odometry& curr_odom,
+	const sensor_msgs::msg::LaserScan& scan)
+{
+    geometry_msgs::msg::PoseWithCovariance dummy_pose{};
```

Comment:
From the comment above on giving the sensor / odometry to the appropriate modules beforehand, this is where when you would implement this that would become more clear. Either way you have to give that info to those modules, so lets push that up into the application so that every solver doesn't need to do that boilerplate (and try to make the solver as independent from those others as possible).